src/Interpreters/AsynchronousMetrics.cpp

```diff
@@ -329,6 +326,45 @@ void AsynchronousMetrics::update()
     saveAllArenasMetric<size_t>(new_values, "muzzy_purged");
 #endif
 
+    // Try to add processor frequencies, ignoring errors.
+    try
+    {
+        ReadBufferFromFile buf("/proc/cpuinfo");
```

Comment:
It allocated a buffer of 1MiB, that's an overkill.
It opens and closes a file everytime... that's also inefficient.

Open once and read with `pread` is better. Look at the `MemoryStatisticsOS`.

Reply:
This is called once a minute, I'd just keep it as simple as possible. Reading with pread is another 50 lines: the size is not known in advance, moreover, we need several read calls (on my laptop, one call returns ~4kB, the entire file is 15kB). Not sure it makes sense.

I can decrease the initial buffer size, there is a parameter for that.

Reply:
Ok, let's just decrease the buffer size slightly (32K is Ok).